daemon/src/monitor.rs

```diff
@@ -207,86 +222,70 @@ impl Cfd {
                     ..self
                 }
             }
-            ContractSetupStarted | ContractSetupFailed | OfferRejected | RolloverRejected => Self {
-                monitor_lock_finality: None,
-                monitor_commit_finality: None,
-                monitor_cet_timelock: false,
-                monitor_refund_timelock: false,
-                monitor_refund_finality: None,
-                monitor_revoked_commit_transactions: Vec::new(),
-                monitor_collaborative_settlement_finality: None,
-                lock_tx: None,
-                cet: None,
-                commit_tx: None,
-                ..self
-            },
             LockConfirmed | LockConfirmedAfterFinality => Self {
-                monitor_lock_finality: None,
-                lock_tx: None,
+                monitor_lock_finality: false,
+                broadcast_lock: None,
                 ..self
             },
             CommitConfirmed => Self {
-                monitor_commit_finality: None,
-                commit_tx: None,
+                monitor_commit_finality: false,
+                broadcast_commit: None,
                 ..self
             },
-            // final states, don't monitor anything
+            // final states, don't monitor or re-broadcast anything
             CetConfirmed | RefundConfirmed | CollaborativeSettlementConfirmed => Self {
-                monitor_lock_finality: None,
-                monitor_commit_finality: None,
+                monitor_lock_finality: false,
+                monitor_commit_finality: false,
                 monitor_cet_timelock: false,
                 monitor_refund_timelock: false,
-                monitor_refund_finality: None,
+                monitor_refund_finality: false,
                 monitor_revoked_commit_transactions: Vec::new(),
-                monitor_collaborative_settlement_finality: None,
-                monitor_cet_finality: None,
-                lock_tx: None,
-                cet: None,
-                commit_tx: None,
+                monitor_collaborative_settlement_finality: false,
+                monitor_cet_finality: false,
+                broadcast_lock: None,
+                broadcast_cet: None,
+                broadcast_commit: None,
                 ..self
             },
             CetTimelockExpiredPriorOracleAttestation => Self {
                 monitor_cet_timelock: false,
                 ..self
             },
-            CetTimelockExpiredPostOracleAttestation { cet, .. } => Self {
-                cet: Some(cet.clone()),
-                monitor_cet_finality: cet_txid_and_script(cet),
+            CetTimelockExpiredPostOracleAttestation { cet, .. }
+            | OracleAttestedPostCetTimelock { cet, .. } => Self {
+                broadcast_cet: Some(cet.clone()),
+                cet: cet_txid_and_script(cet),
+                monitor_cet_finality: true,
                 monitor_cet_timelock: false,
                 ..self
```

Comment:
🤔 Given that we apply the events in order and some events set `broadcast_...` transactions, shouldn't *all* other events (that don't set `broadcast_...` to `Some`) set the value to `None`? 
Otherwise we might pass on `Some` values unintended.

Reply:
I don't think so.

As an example, if we have a `ManualCommit` event which sets the `broadcast_commit` transaction to `Some` and we get an `OracleAttestedPriorCetTimelock`, we must still broadcast the commit transaction.

I think particularly with the rebroadcasting mechanism we don't need to be that careful anyway. If we fail to rebroadcast because it's already there or because something else ended up happening nothing bad can happen.